apps/admin/src/forms/shipmentOptions/fields/createReceiptCodeField.ts

```diff
@@ -0,0 +1,21 @@
+import {TriState} from '@myparcel-pdk/common';
+import {type InteractiveElementConfiguration} from '@myparcel/vue-form-builder';
+import {CarrierName} from '@myparcel/constants';
+import {type ShipmentOptionsRefs} from '../types';
+import {FIELD_AGE_CHECK, FIELD_RECEIPT_CODE} from '../field';
+import {getFormCarrierName, hasPostNlAgeCheck} from '../../helpers';
+import {createShipmentOptionField} from './createShipmentOptionField';
+
+export const createReceiptCodeField = (refs: ShipmentOptionsRefs): InteractiveElementConfiguration => {
+  return createShipmentOptionField(refs, FIELD_RECEIPT_CODE, {
+    readOnlyWhen: ({form}) => hasPostNlAgeCheck(form),
+
+    afterUpdate({form}, value) {
+      if (!(CarrierName.DhlForYou === getFormCarrierName(form) && TriState.On === value)) {
```

Comment:
Would be good to add a comment explaining why the behaviour is different here for DHL For You